### ISSUE TITLE
sync status reports DB/JSONL clean while tracked issues.jsonl is dirty against Git

### ISSUE THREAD

**Issue by Dicklesworthstone** [Dicklesworthstone/beads_rust#338]
## Summary

In a Git-backed Beads workspace, `br sync --status --json --no-auto-import --no-auto-flush` can report the DB and JSONL stores as fully synchronized while the tracked `.beads/issues.jsonl` file is still dirty against Git.

For agent workflows, those are different kinds of cleanliness:

- DB-vs-JSONL clean means Beads' internal stores agree.
- Git-export clean means the tracked JSONL source-of-truth has no uncommitted tracker change.

Today the status payload exposes the first but not the second. In a crowded multi-agent checkout, that makes it unsafe for an agent to use `sync --status` as a tracker authority gate before claiming, closing, or committing source work.

## Environment

Observed in `/Users/jemanuel/projects/eidetic_engine_cli` on 2026-06-07.

```text
br --version: br 0.1.13
branch: main
repo head: 61965d13
tracked export: .beads/issues.jsonl
```

The checkout is a live multi-agent workspace, so this report is about the machine contract exposed by `sync --status`, not a claim that this status command itself was the only possible writer. Related write-path issues are tracked separately in #326 and #337.

## Reproduction / evidence from live workspace

Status says the Beads stores are clean/in sync:

```bash
br sync --status --json --no-auto-import --no-auto-flush
```

Observed:

```json
{
  "dirty_count": 0,
  "last_export_time": "2026-06-07T21:48:14.657645+00:00",
  "last_import_time": "2026-06-07T16:22:43.197427-04:00",
  "jsonl_content_hash": "96d32797983f2c995d28efcc25376239df3b0e955a1f053337ce9e836bfb86c7",
  "jsonl_exists": true,
  "jsonl_newer": false,
  "db_newer": false
}
```

Git simultaneously shows the tracked export is modified:

```bash
git status --porcelain=v2 -- .beads
```

Observed:

```text
1 .M N... 100644 100644 100644 106bcd3357b0e03b28583c93c90c07dbd646e0d7 106bcd3357b0e03b28583c93c90c07dbd646e0d7 .beads/issues.jsonl
```

Diff size:

```bash
git diff --numstat -- .beads/issues.jsonl
git diff --shortstat -- .beads/issues.jsonl
```

Observed:

```text
1       1       .beads/issues.jsonl
1 file changed, 1 insertion(+), 1 deletion(-)
```

The content change was a legitimate tracker export update: a new progress comment appeared on `bd-1n0np.3.7`, moving the issue's `updated_at` from `2026-06-07T21:37:56.113957Z` to `2026-06-07T21:48:14.409248Z` and adding comment id `11297`.

Hashes confirm that Beads' `jsonl_content_hash` matches the worktree export, not the committed tracked export:

```text
worktree .beads/issues.jsonl sha256:
96d32797983f2c995d28efcc25376239df3b0e955a1f053337ce9e836bfb86c7

HEAD .beads/issues.jsonl sha256:
d2db0dc503abafbd3bf87310b4e91aff102c981d10c1385bb7e5c6e31416a1ca
```

So `sync --status` is correct about DB-vs-JSONL synchronization, but incomplete for Git-backed automation: it blesses the dirty worktree hash as the JSONL content hash and gives no machine-visible indication that the tracked export is uncommitted.

## Expected behavior

A Git-backed Beads status surface should distinguish Beads-store authority from Git-export cleanliness.

Suggested additions to `br sync --status --json`:

```json
{
  "dirty_count": 0,
  "jsonl_newer": false,
  "db_newer": false,
  "git_export": {
    "tracked": true,
    "worktree_clean": false,
    "index_clean": true,
    "head_hash": "d2db0dc503abafbd3bf87310b4e91aff102c981d10c1385bb7e5c6e31416a1ca",
    "worktree_hash": "96d32797983f2c995d28efcc25376239df3b0e955a1f053337ce9e836bfb86c7",
    "changed_paths": [".beads/issues.jsonl"]
  },
  "tracker_authority": {
    "stores_in_sync": true,
    "export_committed": false,
    "safe_for_writes": false,
    "safe_for_source_commit": false,
    "reasons": ["tracked_jsonl_dirty_against_git"]
  }
}
```

The exact field names are flexible. The critical invariant is that a robot caller can tell the difference between:

1. DB and JSONL disagree.
2. DB and JSONL agree, but the tracked JSONL export has uncommitted changes.
3. DB, JSONL, and Git are all clean enough for claim/update/source-commit workflows.

## Why this matters

Agents use Beads in shared Git checkouts. A status payload that says `dirty_count=0`, `jsonl_newer=false`, and `db_newer=false` can look safe, but a dirty tracked `.beads/issues.jsonl` still has concrete workflow impact:

- Source commits can accidentally mix tracker-export changes with code/docs changes.
- Claim/close operations may race with an uncommitted tracker update from another agent.
- A conservative agent has to shell out to Git separately and manually reconcile the contradiction.
- Beads and BV automation cannot expose one reliable `safe_for_writes` or `safe_for_commit` field without duplicating Git-export checks.

This is especially important because the current repo already has related Beads reports:

- #326: read/list/status paths can rewrite `issues.jsonl`.
- #337: commit/hook path can silently add unstaged `issues.jsonl` churn.
- #334: umbrella contract for tracker authority and per-issue claimability.

This issue is narrower: even when Beads' internal stores are in sync, `sync --status` should report whether the tracked export is clean against Git.

## Suggested acceptance tests

1. Fixture repo with DB and JSONL synchronized, but `.beads/issues.jsonl` modified relative to Git HEAD.
   - Expected: `stores_in_sync=true`, `git_export.worktree_clean=false`, `safe_for_source_commit=false`.
2. Fixture repo with DB dirty relative to JSONL, but Git export clean.
   - Expected: `stores_in_sync=false`, Git export state reported separately.
3. Fixture repo where DB, JSONL, and Git export are all clean.
   - Expected: `stores_in_sync=true`, `git_export.worktree_clean=true`, `safe_for_writes=true` subject to normal issue-level rules.
4. If the current project is not inside a Git repo, the status should explicitly report `git_export.tracked=false` / `git_export.available=false` rather than omitting the distinction silently.

**Comment by Dicklesworthstone** [comment]
Implemented (simplified scope) in ac04f9c5.

`br sync --status --json` now carries a `git_export` block: `{available, tracked, worktree_clean, index_clean, head_hash, worktree_hash}` — computed from `git status --porcelain` on the JSONL plus the committed blob hash vs `git hash-object` of the worktree file, degrading to `{available: false}` outside a git repo or on any git error (never failing the command). That makes "stores in sync but export not committed" directly visible to agents, closing the invariant gap you described.

Deliberately NOT implemented: the `tracker_authority.safe_for_writes` synthesis — that's a cross-tool contract design (see #334, closed with the same simplification rationale) and the part where the scope creep lives. The raw git facts are now on the wire; the verdict stays with the consumer.
